tests/unit_tests/test_mesh.py

```diff
@@ -397,6 +400,85 @@ def test_umesh_roundtrip(run_in_tmpdir, request):
 
     assert umesh.id == xml_mesh.id
 
+skip_if_no_dagmc = pytest.mark.skipif(
+    not openmc.lib._dagmc_enabled(),
+    reason="DAGMC CAD geometry is not enabled.")
+
+@skip_if_no_dagmc
+@pytest.mark.parametrize('export_type', ('.vtk','.vtu'))
+def test_umesh(request, export_type):
+    """Performs a minimal UnstructuredMesh simulation, reads in the resulting
+    statepoint file and writes the mesh data to vtk and vtkhdf files. It is
+    necessary to read in the unstructured mesh from a statepoint file to ensure
+    it has all the required attributes
+    """
+    surf1 = openmc.Sphere(r=1.0, boundary_type="vacuum")
+    material1 = openmc.Material()
+    material1.add_components(components={"H" : 1.0})
+    material1.set_density('g/cm3', 1.0)
+
+    materials = openmc.Materials([material1])
+    cell1 = openmc.Cell(region=-surf1, fill=material1)
+    geometry = openmc.Geometry([cell1])
+
+    umesh = openmc.UnstructuredMesh(
+       filename= request.path.parent.parent
+        / "regression_tests/unstructured_mesh/test_mesh_dagmc_tets.vtk",
+       library= "moab",
+       mesh_id = 1
+    )
+    # setting ID to make it easier to get the mesh from the statepoint later
+    mesh_filter = openmc.MeshFilter(umesh)
+
+    # Create flux mesh tally to score alpha production
+    mesh_tally = openmc.Tally(name="test_tally")
+    mesh_tally.filters = [mesh_filter]
+    mesh_tally.scores = ["total"]
+
+    tallies = openmc.Tallies([mesh_tally])
+
+    source = openmc.IndependentSource()
+
+    settings = openmc.Settings()
+    settings.run_mode = "fixed source"
+    settings.batches = 2
+    settings.particles = 100
+    settings.source = source
+
+    model = openmc.Model(
+        materials=materials, geometry=geometry, settings=settings, tallies=tallies
+    )
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+
+        statepoint_file = model.run(cwd=tmpdir)
+
+        statepoint = openmc.StatePoint(statepoint_file)
+        tally: openmc.Tally = statepoint.get_tally(name="test_tally")
+        umesh_from_sp: openmc.UnstructuredMesh = statepoint.meshes[1]
```

Comment:
Typically best to open the statepoint files in a context manager to avoid an occupied HDF5 file handle:

```suggestion
        with openmc.StatePoint(statepoint_file) as statepoint:
            tally: openmc.Tally = statepoint.get_tally(name="test_tally")
            umesh_from_sp: openmc.UnstructuredMesh = statepoint.meshes[1]
```